lib/commanded/process_managers/process_manager_instance.ex

```diff
@@ -184,45 +191,56 @@ defmodule Commanded.ProcessManagers.ProcessManagerInstance do
 
     %State{idle_timeout: idle_timeout} = state
 
+    telemetry_metadata = telemetry_metadata(event, state)
+    start_time = telemetry_start(telemetry_metadata)
+
     case handle_event(event, state) do
-      {:error, _error} = error ->
+      {:error, error} ->
         failure_context = build_failure_context(event, context, nil, state)
 
-        handle_event_error(error, event, failure_context, state)
+        telemetry_stop(start_time, telemetry_metadata, {:error, error})
+
+        handle_event_error({:error, error}, event, failure_context, state)
 
       {:error, error, stacktrace} ->
         failure_context = build_failure_context(event, context, stacktrace, state)
 
-        handle_event_error({:error, error}, event, failure_context, state)
+        telemetry_stop(start_time, telemetry_metadata, {:error, error, stacktrace})
 
-      {:stop, _error, _state} = reply ->
-        reply
+        handle_event_error({:error, error}, event, failure_context, state)
 
       commands ->
         # Copy event id, as causation id, and correlation id from handled event.
         opts = [causation_id: event_id, correlation_id: correlation_id, returning: false]
 
-        with :ok <- commands |> List.wrap() |> dispatch_commands(opts, state, event) do
-          case mutate_state(event, state) do
-            {:error, error, stacktrace} ->
-              failure_context = build_failure_context(event, context, stacktrace, state)
+        commands
+        |> List.wrap()
+        |> dispatch_commands(opts, state, event)
+        |> case do
+          :ok ->
+            telemetry_stop(start_time, telemetry_metadata, commands)
```

Comment:
I edited this slightly as the previous formatting was tripping my brain for some reason. I can change back if you prefer the `with`